subsys/net/ip/net_pkt.c

```diff
@@ -10,17 +10,19 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#define LOG_MODULE_NAME net_pkt
-#define NET_LOG_LEVEL CONFIG_NET_PKT_LOG_LEVEL
-
 /* This enables allocation debugging but does not print so much output
  * as that can slow things down a lot.
  */
-#if defined(CONFIG_NET_DEBUG_NET_PKT_ALL)
 #undef NET_LOG_LEVEL
+#if defined(CONFIG_NET_DEBUG_NET_PKT_ALL)
 #define NET_LOG_LEVEL 5
```

Comment:
this looks strange as logger highest level is 4.

Reply:
Yes, unfortunately the max level is 4 as in this module we could use 5 :)
The code just sets the level to 5 if extra debugging is enabled, that way we do not need to check the CONFIG_NET_DEBUG_NET_PKT_ALL option when needed.